examples/all-clusters-app/linux/include/diagnostic-logs-provider-delegate-impl.h

```diff
@@ -0,0 +1,72 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/clusters/diagnostic-logs-server/DiagnosticLogsProviderDelegate.h>
+
+#include <fstream>
+#include <map>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace DiagnosticLogs {
+
+/**
+ * The application delegate to statically define the options.
+ */
+
+class LogProvider : public DiagnosticLogsProviderDelegate
+{
+    static LogProvider sInstance;
+
+public:
+    LogProvider(){};
+    ~LogProvider(){};
+    static inline LogProvider & GetInstance() { return sInstance; }
+
+    void SetEndUserSupportLogFilePath(chip::Optional<std::string> logFilePath) { mEndUserSupportLogFilePath = logFilePath; }
+    void SetNetworkDiagnosticsLogFilePath(chip::Optional<std::string> logFilePath) { mNetworkDiagnosticsLogFilePath = logFilePath; }
+    void SetCrashLogFilePath(chip::Optional<std::string> logFilePath) { mCrashLogFilePath = logFilePath; }
+
+    /////////// DiagnosticLogsProviderDelegate Interface /////////
+    CHIP_ERROR StartLogCollection(IntentEnum intent, LogSessionHandle & outHandle, const Optional<uint64_t> & outTimeStamp,
+                                  const Optional<uint64_t> & outTimeSinceBoot) override;
+    CHIP_ERROR EndLogCollection(LogSessionHandle sessionHandle) override;
+    CHIP_ERROR CollectLog(LogSessionHandle sessionHandle, MutableByteSpan & outBuffer, bool & outIsEndOfLog) override;
+    size_t GetSizeForIntent(IntentEnum intent) override;
+    CHIP_ERROR GetLogForIntent(IntentEnum intent, MutableByteSpan & outBuffer, const Optional<uint64_t> & outTimeStamp,
+                               const Optional<uint64_t> & outTimeSinceBoot) override;
+
+private:
+    chip::Optional<std::string> GetFilePathForIntent(IntentEnum intent);
+
+    chip::Optional<std::string> mEndUserSupportLogFilePath;
+    chip::Optional<std::string> mNetworkDiagnosticsLogFilePath;
+    chip::Optional<std::string> mCrashLogFilePath;
+
+    std::map<LogSessionHandle, FILE *> mFiles;
```

Comment:
Should the destructor ensure all items in mFiles are fclosed ? Currently destructor is empty.

Reply:
Done.